src/coordinator/storage/local/config.go

```diff
@@ -0,0 +1,190 @@
+		return c.newClientFromConfig(c.Client, params, custom...)
+	}
+	return c.Client.NewClient(params, custom...)
+}
+
+// ClusterStaticNamespaceConfiguration describes the namespaces in a
+// static cluster.
+type ClusterStaticNamespaceConfiguration struct {
+	Namespace          string              `yaml:"namespace"`
+	StorageMetricsType storage.MetricsType `yaml:"storageMetricsType"`
+	Retention          time.Duration       `yaml:"retention" validate:"nonzero"`
+	Resolution         time.Duration       `yaml:"resolution" validate:"min=0"`
+}
+
+type unaggregatedClusterNamespaceConfiguration struct {
+	client    client.Client
+	namespace ClusterStaticNamespaceConfiguration
+	result    clusterConnectResult
+}
+
+type aggregatedClusterNamespacesConfiguration struct {
+	client     client.Client
+	namespaces []ClusterStaticNamespaceConfiguration
+	result     clusterConnectResult
+}
+
+type clusterConnectResult struct {
+	session client.Session
+	err     error
+}
+
+// NewClusters instantiates a new Clusters instance.
+func (c ClustersStaticConfiguration) NewClusters() (Clusters, error) {
+	var (
+		numUnaggregatedClusterNamespaces int
+		numAggregatedClusterNamespaces   int
+		unaggregatedClusterNamespaceCfg  = &unaggregatedClusterNamespaceConfiguration{}
+		aggregatedClusterNamespacesCfgs  []*aggregatedClusterNamespacesConfiguration
+		unaggregatedClusterNamespace     UnaggregatedClusterNamespaceDefinition
+		aggregatedClusterNamespaces      []AggregatedClusterNamespaceDefinition
+	)
+	for _, clusterCfg := range c {
+		client, err := clusterCfg.newClient(client.ConfigurationParameters{})
+		if err != nil {
+			return nil, err
+		}
+
+		aggregatedClusterNamespacesCfg := &aggregatedClusterNamespacesConfiguration{
+			client: client,
+		}
+
+		for _, n := range clusterCfg.Namespaces {
+			switch n.StorageMetricsType {
+			case storage.UnaggregatedMetricsType:
+				numUnaggregatedClusterNamespaces++
+				if numUnaggregatedClusterNamespaces > 1 {
+					return nil, fmt.Errorf("only one unaggregated cluster namespace  "+
+						"can be specified: specified %d", numUnaggregatedClusterNamespaces)
+				}
+
+				unaggregatedClusterNamespaceCfg.client = client
+				unaggregatedClusterNamespaceCfg.namespace = n
+
+			case storage.AggregatedMetricsType:
+				numAggregatedClusterNamespaces++
+
+				aggregatedClusterNamespacesCfg.namespaces =
+					append(aggregatedClusterNamespacesCfg.namespaces, n)
+
+			default:
+				return nil, fmt.Errorf("unknown storage metrics type: %v",
+					n.StorageMetricsType)
+			}
+		}
+
+		if len(aggregatedClusterNamespacesCfg.namespaces) > 0 {
+			aggregatedClusterNamespacesCfgs =
+				append(aggregatedClusterNamespacesCfgs, aggregatedClusterNamespacesCfg)
+		}
+	}
+
+	if numUnaggregatedClusterNamespaces != 1 {
+		return nil, fmt.Errorf("one unaggregated cluster namespace  "+
+			"must be specified: specified %d", numUnaggregatedClusterNamespaces)
+	}
+
+	// Connect to all clusters in parallel
+	var wg sync.WaitGroup
+	wg.Add(1)
```

Comment:
Worth having`wg.Add(1 + len(aggregatedClusterNamespacesCfgs))` ?

Reply:
I feel its more native to always `wg.Add(...)` right above an operation, that way if you remove an operation, you remove the `wg.Add(...)` right next to it and don't need to jump around the code (less mistakes in general I've found).